### PR TITLE
display single contained exception in excgroups in test summary

### LINKED ISSUE #12943: `=== short test summary info ===` messages are not useful for `ExceptionGroup`s
If there is only a single leaf exception, at any depth, I suggest we produce the short summary message _for that leaf exception_, and append `[single exception in an ExceptionGroup]` to the message.

### PR DESCRIPTION
**Pull request by jakkdl**
fixes #12943 

As `ExceptionInfo` still doesn't have proper support for exception groups this continues upon the hacky solution from #10209, modifying the reprcrash as well.

I find this solution ... incredibly hacky and ugly, and am not a fan that n==1 gets handled but not n==2. But conceptually it just gets really tricky to figure out what info to strip, and how to show that. If we're going with stripping structure & group messages we could extend this solution to something like `[in ExceptionGroup]: ValueError("foo"), TypeError("bar")`.

EDIT: less hacky now that I moved the code to `ExceptionInfo.exconly`

### CODE REVIEW

File: src/_pytest/_code/code.py [hunk 1]
```diff
@@ -589,6 +589,31 @@ def exconly(self, tryshort: bool = False) -> str:
         representation is returned (so 'AssertionError: ' is removed from
         the beginning).
         """
+
+        def _get_single_subexc(
+            eg: BaseExceptionGroup[BaseException],
+        ) -> BaseException | None:
```
Comment (Zac-HD) [inline src/_pytest/_code/code.py:595]: Simpler implementation:
```python
def _get_single_subexc(
    eg: BaseExceptionGroup[BaseException],
) -> BaseException | None:
    if len(eg.exceptions) != 1:
        return None
    if isinstance(e := eg.exceptions[0], BaseExceptionGroup):
        return _get_single_subexc(e)
    return e
```

File: src/_pytest/_code/code.py [hunk 2]
```diff
@@ -589,6 +589,31 @@ def exconly(self, tryshort: bool = False) -> str:
...
+                else:
+                    res = subexc
+            return res
+
+        if (
+            tryshort
+            and isinstance(self.value, BaseExceptionGroup)
+            and (subexc := _get_single_subexc(self.value)) is not None
+        ):
+            return f"[in {type(self.value).__name__}] {subexc!r}"
```
Comment (Zac-HD) [inline src/_pytest/_code/code.py:615]:
```suggestion
            return f"{subexc!r} [single exception in {type(self.value).__name__}]"
```
as discussed in the issue

Comment (jakkdl) [reply src/_pytest/_code/code.py]: fixed, but noted my disagreement in the issue :)

**Review by Zac-HD — APPROVED** [review]
Thanks @jakkdl!

I really appreciate your writeup in https://github.com/pytest-dev/pytest/issues/12943#issuecomment-2493678326, but continue to prefer this style for a few reasons:
- when the nodeid is long and we can get truncation of this short message, I want to keep the leaf-type as the first part of this message.  Generally users who are seeing ExceptionGroups know they're possible, so I'm not too worried about truncating that note away in such cases
- single-leaf groups are qualitatively simpler to think about than multi-leaf groups - the structure of the latter often matters, for example, while the former are basically just extra structure in the traceback - and so discarding the more complex structure feels a bit worse
  - I think we'll probably want to do something better for these cases too at some point, but I don't yet know what and would prefer to wait until we have a proposal that users seem enthusiastic about.

I'm happy with this PR as-is since it seems to me like a clear improvement on the status quo; if nobody else has feedback I'll merge in a few days 🙂

**Pull request merged**